diff --git a/node_modules/cognito-at-edge/dist/index.js b/node_modules/cognito-at-edge/dist/index.js
index 8a3d2da..96acb72 100644
--- a/node_modules/cognito-at-edge/dist/index.js
+++ b/node_modules/cognito-at-edge/dist/index.js
@@ -26,6 +26,7 @@ class Authenticator {
     _cookieSettingsOverrides;
     _logger;
     _jwtVerifier;
+    _redirectPath;
     constructor(params) {
         this._verifyParams(params);
         this._region = params.region;
@@ -54,6 +55,7 @@ class Authenticator {
         this._csrfProtection = params.csrfProtection;
         this._logoutConfiguration = params.logoutConfiguration;
         this._parseAuthPath = (params.parseAuthPath || '').replace(/^\//, '');
+        this._redirectPath = (params.redirectPath || '').replace(/^\//, '');
     }
     /**
      * Verify that constructor parameters are corrects.
@@ -247,7 +249,7 @@ class Authenticator {
      * @param  {String} location Path to redirection.
      * @return Lambda@Edge response.
      */
-    async _getRedirectResponse(tokens, domain, location) {
+    async _getRedirectResponse(tokens, domain, location, keepMethod = false) {
         const decoded = await this._jwtVerifier.verify(tokens.idToken);
         const username = decoded['cognito:username'];
         const usernameBase = `${this._cookieBase}.${username}`;
@@ -288,7 +290,7 @@ class Authenticator {
             cookies.push(cookie_1.Cookies.serialize(`${this._cookieBase}.${csrf_1.PKCE_COOKIE_NAME_SUFFIX}`, '', csrfCookieAttributes), cookie_1.Cookies.serialize(`${this._cookieBase}.${csrf_1.NONCE_COOKIE_NAME_SUFFIX}`, '', csrfCookieAttributes), cookie_1.Cookies.serialize(`${this._cookieBase}.${csrf_1.NONCE_HMAC_COOKIE_NAME_SUFFIX}`, '', csrfCookieAttributes));
         }
         const response = {
-            status: '302',
+            status: keepMethod ? '307' : '302',
             headers: {
                 location: [
                     {
@@ -598,7 +600,7 @@ class Authenticator {
         const cfDomain = request.headers.host[0].value;
         const redirectURI = this._parseAuthPath
             ? `https://${cfDomain}/${this._parseAuthPath}`
-            : `https://${cfDomain}`;
+            : [`https://${cfDomain}`, this._redirectPath].filter((p) => p).join('/');
         try {
             const tokens = this._getTokensFromCookie(request.headers.cookie);
             if (this._logoutConfiguration &&
@@ -630,7 +632,7 @@ class Authenticator {
                         tokens,
                         err,
                     });
-                    return await this._fetchTokensFromRefreshToken(redirectURI, tokens.refreshToken).then((tokens) => this._getRedirectResponse(tokens, cfDomain, request.uri));
+                    return await this._fetchTokensFromRefreshToken(redirectURI, tokens.refreshToken).then((newTokens) => this._getRedirectResponse({ ...newTokens, refreshToken: tokens.refreshToken }, cfDomain, request.uri, true));
                 }
                 else {
                     throw err;
